content/news/2022-09-02-gsoc-final-blog-david-chocholaty.md

```diff
@@ -0,0 +1,685 @@
+in this project was exactly the implementation of the Pitch Shift effect
+using the RubberBand library. The issue occurred soon, that the effect chain
+offers the effect to work only with the fixed size audio chunks. Based on that,
+it is not possible to require an amount of input audio data as needed
+for the RubberBand library “pull model” implementation. The SoundTouch library
+for Pitch Shifting was tested too but produced results with the worse
+audio quality and with the same amount of delay. Based on all
+the discussions, the RubberBand library has shown as the best option
+for the effect, despite the issues which are associated with it. After that,
+the Mixxx application did not have implemented the effects delay handler
+for the effect chain, so, the implementation of this structure
+was automatically needed. Based on that, the goals of the project were changed.
+The original proposal contained the Pitch shift effect, and with that
+as a project extension to the project requirements of the Mixxx organization,
+the Auto-tune effect was proposed. After consideration, the Zulip chat survey
+for other Mixxx developers and users was created to be able to vote for possible
+project extensions. Based on the survey results, the project extension goal
+was changed to the implementation and optimization of the group delay handling
+for the effect chain to improve the performance of the Pitch Shift effect
+for the wet/dry and wet+dry mode. As the last challenge, I would like to mention
+the usage of the std::span from the standard library which is supported
+by C++20. Because the Mixxx organization adheres to the own
+[Minimum requirements policy](
+ https://github.com/mixxxdj/mixxx/wiki/Minimum%20requirements%20policy)
+for the Ubuntu LTS, the `EngineEffectsDelay` and `RingDelayBuffer`
+could be merged after the official Ubuntu release was announced in the middle
+of August due to support of C++20.
+
+#### Future work
+Concretely for the Pitch Shift effect, the effect will be improved
+using the “pull model” implementation after the end of the GSoC period.
+With that, the wet/dry mode will be done for the effect too.
+As the next project extensions, based on the survey, the following options
+or features can be added to the Pitch shift effect implementation:
+
+* Auto-tune effect
+* A piano keyboard interface
+* Optimize interface for common controllers
+* CPU load balancing
+* Consider interaction with the main Pitch shifter
+* Expose compensation delay as additional parameter for making
+ funny things without extra CPU cycles
+
+More widely, as the future work for the Mixxx application, the wider support
+for the LV2 standard for effects can be implemented or better, the Carla
+audio plugin host can be introduced in the Mixxx application.
+It will allow users to use their own effects they like directly with the support
+of standards such as  LADSPA, DSSI, LV2, VST2, VST3 and so on,
```

Comment:
```suggestion
It will allow users to use their favorite effects enabled via audio plugin
standards such as  LADSPA, DSSI, LV2, VST2, VST3 and so on,
```